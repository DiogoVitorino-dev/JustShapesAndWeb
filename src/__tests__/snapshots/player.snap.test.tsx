import { render, renderHook } from "@testing-library/react-native";
import { useSharedValue } from "react-native-reanimated";

import Player, {
  PlayerAngle,
  PlayerPosition,
  PlayerSize,
} from "@/models/player";

describe("Player model - Snapshot test", () => {
  it("Should renders correctly", () => {
    let pos: PlayerPosition;
    let angle: PlayerAngle;
    let size: PlayerSize;

    renderHook(() => {
      pos = useSharedValue({ x: 25, y: 50 });
      angle = useSharedValue(90);
      size = useSharedValue({ width: 100, height: 50 });
    });

    //@ts-expect-error
    const tree = render(<Player angle={angle} position={pos} size={size} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
