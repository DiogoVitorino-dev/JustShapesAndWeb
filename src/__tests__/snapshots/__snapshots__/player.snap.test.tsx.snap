// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player model - Snapshot test Should renders correctly 1`] = `
<View
  animatedStyle={
    {
      "value": {
        "height": 50,
        "left": 25,
        "top": 50,
        "transform": [
          {
            "rotate": "90deg",
          },
          {
            "scaleY": 1,
          },
          {
            "scale": 1,
          },
        ],
        "width": 100,
      },
    }
  }
  collapsable={false}
  style={
    {
      "backgroundColor": "rgb(20, 255, 170)",
      "borderRadius": 2,
      "height": 50,
      "left": 25,
      "position": "absolute",
      "top": 50,
      "transform": [
        {
          "rotate": "90deg",
        },
        {
          "scaleY": 1,
        },
        {
          "scale": 1,
        },
      ],
      "width": 100,
    }
  }
  testID="playerModel"
/>
`;
